src/auto/async_result.rs

```diff
@@ -0,0 +1,59 @@
+// This file was generated by gir (5a68ad0) from gir-files (469db10)
+// DO NOT EDIT
+
+#[cfg(any(feature = "v2_34", feature = "dox"))]
+use Error;
+use ffi;
+use glib;
+use glib::object::IsA;
+use glib::translate::*;
+use glib_ffi;
+use gobject_ffi;
+use std::mem;
+use std::ptr;
+
+glib_wrapper! {
+    pub struct AsyncResult(Object<ffi::GAsyncResult, ffi::GAsyncResultIface>);
```

Comment:
This is not too useful as we hide the `AsyncResult` from the API elsewhere, in all the async functions. As such this will need some closer integration than this autogenerated API. You can currently never get a `AsyncResult` anywhere.